src/tls/acme/dns/cloudflare.rs

```diff
@@ -140,14 +141,19 @@ impl DnsManager for Cloudflare {
             .await
             .context("unable to find records")?;
 
-        // Delete all matching records
+        // Delete all matching TXT records
         for record in resp.result {
-            self.client
-                .request(&DeleteDnsRecord {
-                    zone_identifier: &zone_id,
-                    identifier: &record.id,
-                })
-                .await?;
+            if let DnsContent::TXT { .. } = &record.content {
+                if record.name.starts_with("_acme-challenge") {
```

Comment:
I think this check is excessive since above we anyway search only for `_acme-challenge`-prefixed records (when called from `TokenManagerDns`)

```
    const ACME_RECORD: &str = "_acme-challenge";
...
    async fn unset(&self, zone: &str) -> Result<(), Error> {
        self.manager.delete(zone, ACME_RECORD).await
    }
```